zio-http/src/test/scala/zhttp/http/URLSpec.scala

```diff
@@ -82,5 +82,30 @@ object URLSpec extends ZIOSpecDefault {
           assertTrue(actual == "/list?query=provided&type=builder")
         },
       ),
+      suite("java interop")(
+        test("can not create a java.net.URI from a relative URL") {
+          check(HttpGen.genRelativeURL) { url =>
+            assert(url.toJavaURI)(isNone)
+          }
+        },
+        test("converts a zhttp.http.URL to java.net.URI") {
+          check(HttpGen.genAbsoluteURL) { autoUrl =>
+            // converts an absolute URL with a relative path (non leading slash) to an absolute path
+            // to avoid encoding problems
+            val absoluteUrl =
+              if (autoUrl.path.nonEmpty && !autoUrl.path.leadingSlash)
+                autoUrl.setPath(Path.root ++ autoUrl.path)
+              else
+                autoUrl
+            val url         = URL.fromString(absoluteUrl.encode).getOrElse(absoluteUrl)
+            val uri         = url.toJavaURI
+            val reversedUrl = uri.flatMap(u => URL.fromString(u.toString).toOption)
+
+            assert(reversedUrl)(
+              equalTo(Some(url)),
+            )
+          }
+        },
```

Comment:
I think we are doing too much over here. I think the test should assert for symmetry. Something like 

```scala
val gen: Gen[URL] = ???

check(gen) { url =>
  val httpURLString = url.encode
  val javaURLString = url.toJava.toString
  assertTrue(httpURLString == javaURLString)
}
```

Reply:
That's exactly the reason about this workaround,
right now, URL, does not seems to be symmetric

```scala
val original: String = "https://localhost/?b=a&H=c&b=b"
val target: String = zhttp.http.URL.fromString(original).toOption.get.encode

// val original: String = https://localhost/?b=a&H=c&b=b
// val target: String = https://localhost/?b=a&b=b&H=c

```

Some other problem is relative paths over an absolute location,
encode generates an invalid url:
```scala
val c = zhttp.http.URL(
    zhttp.http.Path.decode("a/b"),
    zhttp.http.URL.Location.Absolute(zhttp.http.Scheme.HTTPS, "localhost", 433)).encode
// val c: String = https://localhost:433a/b

zhttp.http.URL.fromString(c)
// val res0: Either[Exception,zhttp.http.URL] = Left(java.net.MalformedURLException: Invalid URL: "https://localhost:433a/b")
```	
and some automatically generated urls has this very shape.

Should we consider this:
 * and encode problem
 * An incosistent combination of path and location
 * return 'None' or 'Either.left' in toJavaURI method

Reply:
In that case, we can create a `Gen` from an iterable and manually specify the input cases.

Reply:
Seems that java.net.URI inner parser takes into account all these cases, updated too.